server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -0,0 +1,309 @@
+                            key.reset();
+
+                            enableSettingsWatcher(settingsDir);
+
+                            if (watchedFileChanged(path)) {
+                                processFileSettings(path, false);
+                            }
+                        } catch (Exception e) {
+                            logger.warn("unable to watch or read operator settings file", e);
+                        }
+                    } else {
+                        key.pollEvents();
+                        key.reset();
+                    }
+                }
+            } catch (Exception e) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("encountered exception watching", e);
+                }
+                logger.info("shutting down watcher thread");
+            } finally {
+                watcherThreadLatch.countDown();
+            }
+        }, "elasticsearch[file-settings-watcher]").start();
+    }
+
+    synchronized void stopWatcher() {
+        logger.debug("stopping watcher ...");
+        if (watching()) {
+            try {
+                watchService.close();
+                if (watcherThreadLatch != null) {
+                    watcherThreadLatch.await();
+                }
+            } catch (IOException | InterruptedException e) {
+                logger.info("encountered exception while closing watch service", e);
+            } finally {
+                watchService = null;
+                logger.info("watcher service stopped");
+            }
+        } else {
+            logger.debug("file settings service already stopped");
+        }
+    }
+
+    private void enableSettingsWatcher(Path settingsDir) throws IOException {
+        settingsDir.register(
+            watchService,
+            StandardWatchEventKinds.ENTRY_MODIFY,
+            StandardWatchEventKinds.ENTRY_CREATE,
+            StandardWatchEventKinds.ENTRY_DELETE
+        );
+    }
+
+    void processFileSettings(Path path, boolean onStartup) {
+        logger.info("processing path [{}] for [{}]", path, NAMESPACE);
+        try (
+            XContentParser parser = XContentType.JSON.xContent().createParser(XContentParserConfiguration.EMPTY, Files.newInputStream(path))
+        ) {
+            stateService.process(NAMESPACE, parser, (e) -> {
+                if (e != null) {
+                    // If we encountered an exception trying to apply the operator state at
+                    // startup time, we throw an error to force Elasticsearch to exit.
+                    if (onStartup) {
+                        throw new OperatorConfigurationError("Error applying operator settings", e);
+                    } else {
+                        logger.error("Error processing operator settings json file", e);
+                    }
+                }
+            });
+        } catch (Exception e) {
+            logger.error("Error processing operator settings json file", e);
```

Comment:
process doesn't throw anything checked. So why is this guarded? There is already a catch outside of this method that should handled logging the exception?